packages/core/src/layout/SignInPage/SignInPage.tsx

```diff
@@ -90,31 +90,43 @@ export const SingleSignInPage = ({
   const [retry, setRetry] = useState<{} | boolean | undefined>(auto);
   const [error, setError] = useState<Error>();
 
+  // The SignIn component takes some time to decide whether the user is logged-in or not.
+  // showLoginPage is used to prevent a glitch-like experience where the sign-in page is
+  // displayed for a split second when the user is already logged-in.
+  const [showLoginPage, setShowLoginPage] = useState<boolean>(false);
+
   useEffect(() => {
     const login = async () => {
-      const identity = await authApi.getBackstageIdentity({
-        instantPopup: true,
```

Comment:
We do still want `instantPopup` to be set, to avoid the extra dialog. The missing piece here is to do an initial check with `optional: true`, which will succeed if we already have a session but not return anything if we don't.

Reply:
Wohoo, thanks for the `optional: true` tip - works great 😄